bndtools.m2e/src/bndtools/m2e/BndConfigurator.java

```diff
@@ -329,21 +334,40 @@ private void execJarMojo(final IMavenProjectFacade projectFacade, IProgressMonit
 			}
 
 			// We can now decorate based on the build we just did.
-			try {
-				IProjectDecorator decorator = Injector.ref.get();
-				if (decorator != null) {
-					BndProjectInfo info = new MavenProjectInfo(mavenProject);
-					decorator.updateDecoration(projectFacade.getProject(), info);
-				}
-			} catch (Exception e) {
-				logger.logError("Failed to decorate project " + projectFacade.getProject()
-					.getName(), e);
-			}
+			scheduleDecorate(projectFacade);
 
 			return null;
 		}, monitor);
 	}
 
+	public void scheduleDecorate(final IMavenProjectFacade projectFacade) {
+		final IProject project = projectFacade.getProject();
+
+		Job job = new WorkspaceJob("Executing " + project.getName() + " Bndtools decoration") {
+			@Override
+			public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
+				try {
+					SubMonitor progress = SubMonitor.convert(monitor, 1);
+
+					MavenProject mavenProject = getMavenProject(projectFacade, monitor);
```

Comment:
Should `progress` be used instead of `monitor` as the argument here? I guess an alternate question is whether we need a SubMonitor here?

Reply:
yeah, you're right. Fixing.